### DISCUSSION TITLE
Create a print button?

### DISCUSSION THREAD

**Discussion by anayrat** [sqlpage/SQLPage#499] [Q&A]
Hello,
How can I create a print button ?
I tried with
```sql
select
    'button' as component,
    'lg'     as size;
select 
    'javascript:if(window.print)window.print()'     as link,
    'azure' as outline,
    'Imprimer'  as title,
    'printer'  as icon;
```

But I have this message in the console : 
> Content-Security-Policy: The page’s settings blocked an inline script (script-src-elem) from being executed because it violates the following directive: “script-src 'self' https://cdn.jsdelivr.net”
> Source: javascript:if(window.print)window.print(… create-permanent-user.sql

**Comment by lovasoa** [comment]
Hello! you can use the JavaScript property of the shell component to include a custom script.

**Comment by lovasoa** [comment]
Hello ! Just a quick update for anyone stumbling on this now: 

Since v0.25, SQLPage allows customizing the content-security policy. The default one is still restrictive, but you can disable it by setting

```json
{ "content_security_policy": "" }
```

in [`sqlpage/sqlpage.json`](https://github.com/lovasoa/SQLpage/blob/main/configuration.md)
